src/Symfony/Component/HttpKernel/Controller/ErrorController.php

```diff
@@ -30,34 +29,30 @@ class ErrorController
     private $controller;
     private $errorRenderer;
 
-    public function __construct(HttpKernelInterface $kernel, $controller, ErrorRenderer $errorRenderer)
+    public function __construct(HttpKernelInterface $kernel, $controller, ErrorRendererInterface $errorRenderer)
     {
         $this->kernel = $kernel;
         $this->controller = $controller;
         $this->errorRenderer = $errorRenderer;
     }
 
-    public function __invoke(Request $request, FlattenException $exception): Response
+    public function __invoke(\Throwable $exception): Response
     {
-        try {
-            return new Response($this->errorRenderer->render($exception, $request->getPreferredFormat()), $exception->getStatusCode(), $exception->getHeaders());
-        } catch (ErrorRendererNotFoundException $_) {
-            return new Response($this->errorRenderer->render($exception), $exception->getStatusCode(), $exception->getHeaders());
-        }
+        $exception = $this->errorRenderer->render($exception);
+
+        return new Response($exception->getAsString(), $exception->getStatusCode(), $exception->getHeaders());
     }
 
     public function preview(Request $request, int $code): Response
     {
-        $exception = FlattenException::createFromThrowable(new \Exception('This is a sample exception.'), $code, ['X-Debug' => false]);
-
         /*
          * This Request mimics the parameters set by
          * \Symfony\Component\HttpKernel\EventListener\ErrorListener::duplicateRequest, with
          * the additional "showException" flag.
          */
         $subRequest = $request->duplicate(null, null, [
             '_controller' => $this->controller,
-            'exception' => $exception,
+            'exception' => new HttpException($code, 'This is a sample exception.'),
             'logger' => null,
             'showException' => false,
```

Comment:
it's missing use this attribute to show the non-debug version of the response on e.g. `/__error/404.html` or `/__error/404.json`.

Reply:
However, I'm not sure it's especially useful for non-HTML formats.

Reply:
I'll let you follow up on this topic after merge if you don't mind.

Reply:
Done in https://github.com/symfony/symfony/pull/34331